prow/plugins/config.go

```diff
@@ -1786,9 +1786,58 @@ func (c *Configuration) mergeFrom(other *Configuration) error {
 		errs = append(errs, fmt.Errorf("failed to merge .bugzilla from supplemental config: %w", err))
 	}
 
+	if err := c.mergeApproveFrom(other.Approve, org, repo); err != nil {
+		errs = append(errs, fmt.Errorf("failed to merge .approve from supplemental config: %w", err))
+	}
+
 	return utilerrors.NewAggregate(errs)
 }
 
+func (c *Configuration) mergeApproveFrom(other []Approve, org, repo string) error {
+	if len(other) == 0 {
+		return nil
+	}
+
+	// If the supplemental configuration points to a specific org/repo, we need to validate that there is
+	// only one approve configuration that holds that org/repo value in the `repos` field.
+	if org != "" && repo != "" {
+		if len(other) > 1 {
+			return fmt.Errorf("supplemental config for %s/%s must have only 1 approve plugin configuration", org, repo)
+		}
+		for _, approve := range other {
+			if len(approve.Repos) != 1 || (len(approve.Repos) == 1 && approve.Repos[0] != fmt.Sprintf("%s/%s", org, repo)) {
+				return fmt.Errorf("the repos field must hold the %s/%s value only", org, repo)
+			}
+		}
+	}
+
+	// If the supplemental configuration points to an organization, we need to validate that each of the configurations holds
```

Comment:
Hm - values for `orgA/repoB` should not be in the `orgA` file - they should be in the repo-specific file. We should instead validate that an org-level file has *only* org level configuration

Reply:
What about adding the ability to have an org-level config that affects only specific repos that belong in that org?

Reply:
> What about adding the ability to have an org-level config that affects only specific repos that belong in that org?

That makes it repo specific and not org-specific anymore so has to be in repo config files

Reply:
Thanks for the clarification. I'll make the changes.